### PR TITLE
Me people spatula

### PR DESCRIPTION
**Pull request by morden35**
Writing ME people scraper using spatula framework. Notes:

1. Senate list page source is an excel file: https://legislature.maine.gov/uploads/visual_edit/130th-senate-contact-information-as-of-31221.xlsx. Using ExcelListPage.

2. Senate list page source has 'Mailing Address'. Storing this in p.extras. Could alternatively be a district office address.

3. Appending "(207)" to front of Senate phones.

4. Senate detail pages also list an address. If this address is not the same as 'Mailing Address', I am storing in p.extras['Additional address']. Could alternatively be a district office address or p.additional_offices.

5. No capitol_office.address information or district_office.address for Senators. Just a 'Mailing Address' and 'Additional Address' in p.extras.

6. Skipping over https://legislature.maine.gov/house/house/MemberProfiles/Details/1361 because he resigned on 7/4/21.

7. I used _party_map dictionary to assign parties to representatives. Does it look accurate?
_party_map = {
    "D": "Democratic",
    "R": "Republican",
    "I": "Independent",
    "L": "Libertarian",
    "Passamaquoddy Tribe": "Independent",
}

8. https://legislature.maine.gov/house/house/MemberProfiles/Details/1193 does not have a district listed. She also belongs to the 'Passamaquoddy Tribe' party which I assigned as "Independent".

@jamesturk

### CODE REVIEW

File: scrapers_next/me/people.py [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+
+
+@dataclass
+class PartialPerson:
+    name: str
+    party: str
+    source: str
+
+
+# is this correct?
```
Comment (jamesturk) [inline scrapers_next/me/people.py:22]: looks good to me, the Passamaquoddy seems like a special case & I'm good with marking it independent

File: scrapers_next/me/people.py [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+                .match_one(self.root)
+                .getnext()
+            )
+            p.extras["occupation"] = occupation.text_content().strip()
+        except SelectorError:
+            pass
+
+        try:
+            if (
+                XPath(
```
Comment (jamesturk) [inline scrapers_next/me/people.py:143]: I think the repeated XPath here could be replaced with 

education = Xpath(...)... 
if education is not None:
   p.extras["education"] = education...

File: scrapers_next/me/people.py [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+        cap_addr = XPath(
+            "//*[@id='welcome']/div/div[1]/div/div/table/tbody/tr[3]/td[2]/text()"
+        ).match(self.root)
+        capitol_address = ""
+        for line in cap_addr:
+            capitol_address += line.strip()
+            capitol_address += " "
+        p.capitol_office.address = capitol_address.strip()
+
+        # education
```
Comment (jamesturk) [inline scrapers_next/me/people.py:128]: see below comment on duplicate XPaths

File: scrapers_next/me/people.py [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+        website = (
+            XPath("//*[@id='content']/p/strong[contains(text(), 'Website')]")
+            .match_one(self.root)
+            .getnext()
+        )
+        if website.get("href") is None:
+            website = website.getnext().get("href")
+        else:
+            website = website.get("href")
+        p.extras["website"] = website
```
Comment (jamesturk) [inline scrapers_next/me/people.py:260]: website can be added via add_link instead of on extras :)

File: scrapers_next/me/people.py [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+            address = re.sub(r"Saint\s", "St. ", address)
+        if re.search(r"N\.", address):
+            address = re.sub(r"N\.", "North", address)
+        p.extras["Mailing address"] = address
+        # should this be a district office?
+
+        phone = item[9].strip()
+        phone = "(207) " + phone
+        # append area code to phone?
+        p.extras["phone"] = phone
```
Comment (jamesturk) [inline scrapers_next/me/people.py:319]: any reason for these addresses/phone #s  to be on extras?

**Review by jamesturk — CHANGES_REQUESTED** [review]

File: scrapers_next/me/people.py (hunk 5 above)
Comment (morden35) [reply scrapers_next/me/people.py:319]: Wasn't sure if these should be assigned to capitol or district offices. I can change to district.

**Pull request merged**